src/soc/ibm/power9/istep_8_10.c

```diff
@@ -0,0 +1,1449 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+
+#include <arch/byteorder.h>
+#include <console/console.h>
+#include <cpu/power/istep_8_10.h>
+#include <cpu/power/scom.h>
+#include <cpu/power/scom_registers.h>
+#include <timer.h>
+#include <types.h>
+
+void istep_8_10(void)
+{
+    printk(BIOS_EMERG, "starting istep 8.10\n");
+    report_istep(8, 10);
+    // p9_io_xbus_scominit(0);
+    // p9_io_xbus_scominit(1);
+    printk(BIOS_EMERG, "ending istep 8.10\n");
+    return;
+}
+
+void p9_io_xbus_scominit(const uint8_t group)
+{
+    scom_or_for_chiplet(XB_CHIPLET_ID, 0x8009F8000000003F, PPC_BIT(52));
+    wait_us(50, false);
+    scom_or_for_chiplet(XB_CHIPLET_ID, 0x800C9C000000003F, PPC_BIT(48));
+    wait_us(50, false);
+
+    if(0 == group)
+    {
+        p9_xbus_g0_scom();
+    }
+    else if(1 == group)
+    {
+        p9_xbus_g1_scom();
+    }
+
+    if(!(read_scom(PB_CENT_SM0_PB_CENT_FIR_REG) & PPC_BIT(13)))
+    {
+        write_scom_for_chiplet(XB_CHIPLET_ID, XBUS_FIR_ACTION0_REG, XBUS_PHY_FIR_ACTION0);
+        write_scom_for_chiplet(XB_CHIPLET_ID, XBUS_FIR_ACTION1_REG, XBUS_PHY_FIR_ACTION1);
+        write_scom_for_chiplet(XB_CHIPLET_ID, XBUS_FIR_MASK_REG, XBUS_PHY_FIR_MASK);
+    }
+}
+
+void p9_xbus_g0_scom()
+{
+    // register P9A_XBUS_0_RX0_RXPACKS3_SLICE4_RX_DATA_DAC_SPARE_MODE_PL
+    // IOF1_RX_RX0_RXPACKS_3_RXPACK_RD_SLICE_4_RX_DAC_REGS_RX_DAC_REGS_RX_PL_DATA_DAC_SPARE_MODE_5_OFF
+    // IOF1_RX_RX0_RXPACKS_3_RXPACK_RD_SLICE_4_RX_DAC_REGS_RX_DAC_REGS_RX_PL_DATA_DAC_SPARE_MODE_6_OFF
+    // IOF1_RX_RX0_RXPACKS_3_RXPACK_RD_SLICE_4_RX_DAC_REGS_RX_DAC_REGS_RX_PL_DATA_DAC_SPARE_MODE_7_OFF
+    scom_and_for_chiplet(XB_CHIPLET_ID, 0x8000000006010C3F, ~PPC_BITMASK(53, 55));
```

Comment:
Zamiast adresu hardcodowanego dajmy define P9A_XBUS_0_RX0_RXPACKS3_SLICE4_RX_DATA_DAC_SPARE_MODE_PL
i usuńmy te komentraze ponad. Dodatkowo widzę, że wpisujemy do tego samego rejestru ale od bitu 32 jest jakiś index. Idealnie gdyby było `P9A_XBUS_0_RX0_RXPACKS3_SLICE4_RX_DATA_DAC_SPARE_MODE_PL | (ID << 32)`. I to do pętli FOR

I tak dla wszystkich linii poniżej.

Reply:
I have cleand it up, and wrote a loop aroud register writes, of values which are the same for each id